### ISSUE TITLE
Attachments Processed Multiple Times

### ISSUE THREAD

**Issue by ihennessey** [ahochsteger/gmail2gdrive#11]
Can anyone explain why this script pulls attachments from emails that have already been processed?  

It is correctly saving attachments according to my filters, but also saves attachments from emails that match my subject filter that have been trashed in Gmail even when I have removed the label from these emails.

**Comment by ahochsteger** [comment]
@ihennessey you can prevent searching in trash by adding '-in:trash' to your filter rules.

**Comment by ahochsteger** [comment]
@ihennessey FYI, I've just added a new global filter to the config that excludes several folders by default (e.g. trash, spam, drafts). May you update to the latest version and tell me if it works for you?

**Comment by ihennessey** [comment]
Thank you @ahochsteger,

Since requesting help, I did add filters to exclude labels, but what is happening is that the script is pulling all messages in the thread even if all but one message is deleted.  I am learning, but I don't yet know how to limit the loop to only select the newest message in the thread.  Attached is the log output showing the messages being processed.   All the messages in this example were already deleted except for the last message in the log. The email that contained attachment 032486.xls.

![screen shot 2017-03-26 at march 269 04 49 am](https://cloud.githubusercontent.com/assets/26444135/24332947/5fc9b806-1204-11e7-83cc-c0207c89e2d9.png)

**Comment by ahochsteger** [comment]
@ihennessey I think the behavior is that way, because the GMail API always returns the whole thread instead of individual messages. Also the tagging of processed emails is done on the thread level (which might be possible to do on the message level according to the API spec).

**Comment by ilmike06** [comment]
## Hi @ahochsteger 
Thank you for the script and detailed instructions. They are very good!
I have the same issue with my use case. I'm wondering if the advance gmail functions where a message/thread can be deleted (forever) could be used as a work around. I'm fine with **permanently deleting the email** once the file has been downloaded as a way to get around the thread limitation. I've tried it manually and the it works. I just need to figure out how to make the script permanently delete the thread after processing. I don't know how to implement the code found here: [developers.google.com/gmail/api/v1/reference/users/messages/delete](https://developers.google.com/gmail/api/v1/reference/users/messages/delete)
Can you you give some advise on how to implement this code into your script?

**Comment by ilmike06** [comment]
To follow up on my own question, I was able to get it to work with reading some of the other responses more carefully.

I used this function and inserted the call to this function after the moving the thread to the trash within the "// Mark a thread as processed:" section of the Gmail2GDrive function.
Also had to enable the Gmail API in the Advance Google Services section.

`function deleteForever() {

 var labelName = "My-Label-To-Be-Deleted "

  var threads = GmailApp.search("in:trash label:" + labelName);
  for (var i = 0; i < threads.length; i++) { 
    Gmail.Users.Threads.remove('myuserID', threads[i].getId());
  }
}`
